GitUI/UserControls/RevisionGrid/ChangeCount.cs

```diff
@@ -0,0 +1,91 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Reactive.Linq;
+using System.Text;
+using GitCommands;
+
+namespace GitUI
+{
+    public class ChangeCount
+    {
+        // Count for artificial commits
+        public IReadOnlyList<GitItemStatus> Changed { get; set; }
+        public IReadOnlyList<GitItemStatus> New { get; set; }
+        public IReadOnlyList<GitItemStatus> Deleted { get; set; }
+        public IReadOnlyList<GitItemStatus> SubmodulesChanged { get; set; }
+        public IReadOnlyList<GitItemStatus> SubmodulesDirty { get; set; }
+
+        public bool HasChanges
+            => (Changed?.Count ?? 0) > 0
+               || (New?.Count ?? 0) > 0
+               || (Deleted?.Count ?? 0) > 0
+               || (SubmodulesChanged?.Count ?? 0) > 0
+               || (SubmodulesDirty?.Count ?? 0) > 0;
+
+        public void UpdateChangeCount(IReadOnlyList<GitItemStatus> items)
+        {
+            Changed = items.Where(item => !item.IsNew && !item.IsDeleted && !item.IsSubmodule).ToList();
+            New = items.Where(item => item.IsNew && !item.IsSubmodule).ToList();
+            Deleted = items.Where(item => item.IsDeleted && !item.IsSubmodule).ToList();
+            SubmodulesChanged = items.Where(item => item.IsSubmodule && item.IsChanged).ToList();
+            SubmodulesDirty = items.Where(item => item.IsSubmodule && item.IsDirty).ToList();
+        }
+
+        public StringBuilder GetSummary()
+        {
+            StringBuilder builder = new StringBuilder();
+
+            // TODO use translation strings here
+            Append(Changed, "changed file");
+            Append(Deleted, "deleted file");
+            Append(New, "new file");
+            Append(SubmodulesChanged, "changed submodule");
+            Append(SubmodulesDirty, "dirty submodule");
+
+            return builder;
+
+            void Append(IReadOnlyList<GitItemStatus> items, string singular)
+            {
+                if (items == null || items.Count == 0)
+                {
+                    return;
+                }
+
+                if (builder.Length != 0)
+                {
+                    builder.AppendLine();
+                }
+
+                builder.Append(items.Count).Append(' ');
+
+                if (items.Count == 1)
+                {
+                    builder.AppendLine(singular);
+                }
+                else
+                {
+                    builder.Append(singular).AppendLine("s");
+                }
```

Comment:
```
var ending = string.Empty;

if (items.Count > 1)
{
    ending = "s";
}

builder.AppendFormat("{0} {1}{2}", items.Count, singular, ending)
    .AppendLine();
```

Reply:
If this is to be changed interpolated string should be used instead
````
                builder.Append($"{singular}{(items.Count == 1 ? "" : "s")} {items.Count}")
                    .AppendLine();
``

Reply:
A new string object will be created every time.

Reply:
![image](https://user-images.githubusercontent.com/3169265/103888635-b8a19300-50ed-11eb-9541-ddf666ae55f9.png)